README.md

```diff
@@ -73,6 +73,24 @@ This directory can contain additional postgresql settings in `*.conf` files.
 Copy `auth_root_page.html` to `/home/reverse-proxy/` or wherever your `.env`
 expects to find it via `AUTH_ROOT_PAGE`.
 
+## SMS messages for two-factor authentication
+
+Build a Twilio SMS Keycloak provider jar from `./twilio-keycloak-provider` in:
+https://github.com/PhilanthropyDataCommons/auth
+Set `KEYCLOAK_CUSTOM_SMS_PROVIDER_JAR` in `.env` to the path of this jar file.
+
+Build dasniko's keycloak requiredaction jar from `./requiredaction` in:
+https://github.com/dasniko/keycloak-extensions-demo
+Set `KEYCLOAK_CUSTOM_SMS_PROVIDER_DEPENDENCY_JAR` in `.env` to the path of this
+jar file.
+
+More details can be found in the [auth/twilio-keycloak-provider README](
+https://github.com/PhilanthropyDataCommons/auth/tree/main/twilio-keycloak-provider
+).
+
+Because neither of these jars is published to a central repository, a precompiled
+version of each is included in the lib directory in this repository.
```

Comment:
I appreciate the expedience, but this doesn't sound super sustainable! What process do you propose for updating these? Can we plan on using GitHub Releases or so to host binary assets, or host our own repository, or upload to a central repository?

Reply:
In general the goal would be to publish packages as part of build automation, e.g. https://docs.github.com/en/packages/working-with-a-github-packages-registry, and pull packages with build automation on the consuming side as well. Here (in `deploy`) consumption is a little different from the usual project (e.g. `npm` in a js project), so we would need to think of a way to combine all of the little files together. Perhaps a compressed tarball or `.deb` would make sense. What do you think?

Reply:
> In general the goal would be to publish packages as part of build automation, e.g. https://docs.github.com/en/packages/working-with-a-github-packages-registry, and pull packages with build automation on the consuming side as well.

I haven't used the GitHub Packages Registry, but that looks perfect! Big +1 to the plan of publishing there.

> Here (in `deploy`) consumption is a little different from the usual project (e.g. `npm` in a js project), so we would need to think of a way to combine all of the little files together.

Can you say more about the difference? There's some work to do to fetch the packages and stage them in the way that Keycloak needs, but that matches up well with what I'd imagine is the purpose of this repo.

> Perhaps a compressed tarball or `.deb` would make sense. What do you think?

A `.deb` would make sense if we were using Debian packages to manage Keycloak! Last time I looked there are no upstream packages, but I'd be on board with the idea of packaging Keycloak ourselves.

Overall, though, maybe I'm misunderstanding the problem this would solve - can you say more?